### DISCUSSION TITLE
configuring router client communication

### DISCUSSION THREAD

**Discussion by ros2torial** [eclipse-zenoh/roadmap#82] [Q&A] [answered]
Hi, 
We want to try out a setup for fleet management such that the zenoh router will be in the fleet management PC and zenoh client will be in individual robot. In this case, the individual robot zenoh client shouldn't share any data or meta data with each other. Each robots zenoh client should only communicate and share data with router where a database will be running to store data. 
By default, each zenoh client was sharing meta data(127 bytes) with each other through zenoh router. I experimented with scouting and routing tags in configuration to stop meta data sharing between zenoh clients but it didn't work out. Can anyone please tell if it is possible or not and if it is possible then how to configure it?

**Comment by JEnoch** [comment, marked as answer]
Hi @ros2torial ,

Sure it's possible. You just need to configure your Zenoh clients applications to use the `client` mode.

Most of our bridges, examples and demos supports the command line arguments `-m client` or  `--mode client` to activate this mode.
For your own application, you can have copy or adapt what we do in our examples:
 - in Rust: https://github.com/eclipse-zenoh/zenoh/blob/master/examples/examples/z_put.rs#L66
 - in Python: https://github.com/eclipse-zenoh/zenoh-python/blob/master/examples/z_put.py#L56

By default a Zenoh application runs in `peer` mode, where it tries to establish direct connectivity with each discovered Zenoh application and Zenoh router. Those can be discovered either via scouting (by default over UDP multicast), either via gossip discovery (a router or a peer forwarding the locators from another router/peer).
In `client` mode the Zenoh application establishes only 1 connection with a configured or discovered router (or peer).

**Reply by ros2torial** [reply]
Thanks @JEnoch for the information. I have mentioned my comments in #79 .
